Add tools details to the integration tools section (#9683)
ballerina-platform/ballerina-dev-website: swan-lake/integration-tools/xsd-tool.md

---
layout: ballerina-xsd-support-left-nav-pages-swanlake
title: XSD tool
description: The XSD tool provides the below set of command line tools to work with XSD files in Ballerina.
keywords: ballerina, programming language, xsd, xml
permalink: /learn/xsd-tool/
active: xsd-tool
intro: The XSD tool provides the below set of command line tools to work with XSD files in Ballerina.
--- 

The Ballerina XSD tool generates Ballerina record types from an XSD specification, simplifying integration with XML-based operations in Ballerina.

## Install the tool

Execute the command below to pull the XSD tool from [Ballerina Central](https://central.ballerina.io/ballerina/xsd/latest).

> ```bash

Reviewer: ```suggestion
```
```

$ bal tool pull xsd
```

## Usage

The XSD tool provides the capability to generate the Ballerina record types for a given XSD schema.

### Generate Ballerina clients from XSD Specification

The record types generated from an XSD file can be used in Ballerina applications to handle XML-based data defined in the XSD specification

The following command will generate Ballerina records for a given XSD file.

> ```bash

Reviewer: ```suggestion
```
```

$ bal xsd <xsd-file-name> --module <output-module-name>
```

| Option | Description |
|--------|-------------|
| `<xsd-file-path>` | (Required) The path to the WSDL file. |
|`-m, --module <output-module-name>` |: The name of the module in which Ballerina record types will be generated. If not provided, the output file will be saved to the same Ballerina project. |

For example,

> ```bash

Reviewer: ```suggestion
```
```

$ bal xsd hello.xsd --module custom
```

This generates record types (`types.bal`) for the `hello.xsd` XSD specification. The above command can be run from anywhere on the execution path. It is mandatory to run the above command within a Ballerina package.

Upon successful execution, the following file will be created in the `custom` directory within the Ballerina project.

> ```bash

Reviewer: ```suggestion
```
```

-- types.bal
```

## Examples

### Generate record types in the default module

This command generates record types in the current Ballerina project.

> ```bash

Reviewer: ```suggestion
```
```

$ bal xsd sample.xsd
```

### Generate record types in a specific module

To organize the generated types in a specific module, use the `--module` flag.

> ```bash

Reviewer: ```suggestion
```
```

$ bal xsd sample.xsd --module custom
```

The generated types.bal file will be placed inside the `custom` module.

> ```bash

Reviewer: ```suggestion
```
```

---
layout: ballerina-xsd-support-left-nav-pages-swanlake
title: XSD tool

> description: The XSD tool provides the below set of command line tools to work with XSD files in Ballerina.

Reviewer 2: ```suggestion
description: The XSD tool provides a set of command line tools to work with XSD files in Ballerina.
```

keywords: ballerina, programming language, xsd, xml
permalink: /learn/xsd-tool/
active: xsd-tool

> intro: The XSD tool provides the below set of command line tools to work with XSD files in Ballerina.
> --- 
> 
> The Ballerina XSD tool generates Ballerina record types from an XSD specification, simplifying integration with XML-based operations in Ballerina.
> 

Reviewer 2: ```suggestion
intro: The Ballerina XSD tool generates Ballerina record types from an XSD specification, simplifying integration with XML-based operations in Ballerina.
--- 
```

```
$ bal xsd <xsd-file-path> 
          [--module <output-module-name>]
```

### Command options  

| Option | Description | Mandatory/Optional |
|--------|-------------|--------------------|
| `<xsd-file-path>` | The path of the XSD file | Mandatory |
| `-m`, `--module`   | The name of the module in which the Ballerina record types are generated. If not provided, the output file will be saved to the project default package | Optional |

### Generate types for the given XSD file

Use the following command to generate Ballerina record types for all elements defined in the specified XSD file. By default, the generated `types.bal` file will be placed in the default module of the current Ballerina project.

```
$ bal xsd <source-file-path>
```

For example,

```
$ bal xsd sample.xsd
```

If successful, you will see the following output.

```
The 'types.bal' file is written to the default module
```

### Generate types in a specific module

To generate the Ballerina record types in a specific module, use the `--module` option.

```
$ bal xsd <source-file-path> --module <output-module-name>
```

For example,

```
$ bal xsd sample.xsd --module custom
```

> This will generate a `types.bal` file inside the `modules/custom` directory of the Ballerina project.

Reviewer 2: ```suggestion
This will generate a `types.bal` file inside the `custom` submodule of the Ballerina project.
```
